prusti-tests/tests/typecheck/pass/ghost_int.rs

```diff
@@ -1,9 +1,15 @@
+use prusti_contracts::GhostInt;
 use prusti_contracts::*;
-use crate::ghost::GhostInt;
 
-fn test4() {
-    let a = GhostInt;
-    let b = GhostInt;
-    let c = a + b;
-}   
+fn test_1() -> GhostInt{
+    let x = GhostInt::new(10);
+    let y = GhostInt::new(20);
+    x + y
+}
+
+#[ensures(result == gh1 + gh2)]
+fn test_2(gh1: GhostInt, gh2: GhostInt) -> GhostInt{
```

Comment:
With the specified post-condition, the compiler panics at undefined encoding for Add. 

Does this mean the `add` method is not encoded to viper? But [`encode_bin_op_expr`](https://github.com/viperproject/prusti-dev/blob/1b3fbd6aa76ad8489df08e14f5ee847a159e19cc/prusti-viper/src/encoder/mir_encoder.rs#L416) seem to be performing the necessary encodings for the `BinOp::Add` case.

```
Verification of 3 items...
error: internal compiler error: compiler/rustc_metadata/src/rmeta/decoder.rs:1209:17: get_optimized_mir: missing MIR for `DefId(2:2102 ~ core[5bc6]::ops::arith::Add::add)`
```

Is there something that I'm overlooking here?

Reply:
The entire backtrace:

```
thread 'rustc' panicked at 'Box<Any>', compiler/rustc_errors/src/lib.rs:945:9
stack backtrace:
   0:     0x7f5eab4e16c0 - std::backtrace_rs::backtrace::libunwind::trace::he85dfb3ae4206056
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/../../backtrace/src/backtrace/libunwind.rs:96
   1:     0x7f5eab4e16c0 - std::backtrace_rs::backtrace::trace_unsynchronized::h1ad28094d7b00c21
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:     0x7f5eab4e16c0 - std::sys_common::backtrace::_print_fmt::h901b54610713cd21
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/sys_common/backtrace.rs:79
   3:     0x7f5eab4e16c0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hb0ad78ee1571f7e0
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/sys_common/backtrace.rs:58
   4:     0x7f5eab54f65c - core::fmt::write::h1857a60b204f1b6a
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/core/src/fmt/mod.rs:1080
   5:     0x7f5eab4d39e2 - std::io::Write::write_fmt::hf7b7d7b243f84a36
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/io/mod.rs:1516
   6:     0x7f5eab4e622d - std::sys_common::backtrace::_print::hd093978a5287b8ff
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/sys_common/backtrace.rs:61
   7:     0x7f5eab4e622d - std::sys_common::backtrace::print::h20f46787581d56d7
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/sys_common/backtrace.rs:48
   8:     0x7f5eab4e622d - std::panicking::default_hook::{{closure}}::h486cbb4b82ffc357
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/panicking.rs:208
   9:     0x7f5eab4e5ed8 - std::panicking::default_hook::h4190c9e3edd4d591
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/panicking.rs:227
  10:     0x55a98d264cb6 - <alloc::boxed::Box<F> as core::ops::function::Fn<A>>::call::h08818e2850f53bf9
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1056
  11:     0x55a98d27a3e7 - prusti_driver::report_prusti_ice::h6b050bc319f1869f
                               at /home/aakp/prusti-dev/prusti/src/driver.rs:67
  12:     0x55a98d2738b5 - <prusti_driver::ICE_HOOK as core::ops::deref::Deref>::deref::__static_ref_initialize::{{closure}}::h284b79448c38478d
                               at /home/aakp/prusti-dev/prusti/src/driver.rs:50
  13:     0x7f5e9a8b4fb6 - <alloc::boxed::Box<F> as core::ops::function::Fn<A>>::call::h27e23d762e20efca
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1056
  14:     0x7f5e9a8b6e4b - proc_macro::bridge::client::<impl proc_macro::bridge::Bridge>::enter::{{closure}}::{{closure}}::h2f13c7b6b966cf7a
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/proc_macro/src/bridge/client.rs:320
  15:     0x7f5eab4e6a76 - std::panicking::rust_panic_with_hook::h72e78719cdda225c
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/panicking.rs:581
  16:     0x7f5eaed1d0dd - std::panicking::begin_panic::{{closure}}::h07c37e0eb1cdd0bd
  17:     0x7f5eaed1d006 - std::sys_common::backtrace::__rust_end_short_backtrace::h0e913c6e63ac7532
  18:     0x7f5eaed1753f - std::panicking::begin_panic::hb4984aaf49f869f2
  19:     0x7f5eaed55bd0 - rustc_errors::HandlerInner::bug::hba4e38a3874f5dd8
  20:     0x7f5eaed54780 - rustc_errors::Handler::bug::h9b8941b572d00bc2
  21:     0x7f5eae69f074 - rustc_middle::util::bug::opt_span_bug_fmt::{{closure}}::h87b49eb3e62a46d3
  22:     0x7f5eae69885b - rustc_middle::ty::context::tls::with_opt::{{closure}}::h6b09398a42833be9
  23:     0x7f5eae6981d2 - rustc_middle::ty::context::tls::with_opt::h3ed1d76bd82dcf5d
  24:     0x7f5eae69ef99 - rustc_middle::util::bug::opt_span_bug_fmt::hdd7e8ace9c7ba273
  25:     0x7f5eae69ef0e - rustc_middle::util::bug::bug_fmt::hefa16261d53957ce
  26:     0x7f5eadbcef3e - rustc_metadata::rmeta::decoder::<impl rustc_metadata::creader::CrateMetadataRef>::get_optimized_mir::{{closure}}::h785ede419835474f
  27:     0x7f5eadbcee6c - rustc_metadata::rmeta::decoder::<impl rustc_metadata::creader::CrateMetadataRef>::get_optimized_mir::hd44d7acd74bc94b1
  28:     0x7f5eadb1fd84 - rustc_metadata::rmeta::decoder::cstore_impl::provide_extern::optimized_mir::h2c7732cafeb113b5
  29:     0x55a98d74a0cc - rustc_middle::ty::query::<impl rustc_query_system::query::config::QueryAccessors<rustc_middle::ty::context::TyCtxt> for rustc_middle::ty::query::queries::optimized_mir>::compute::hb094b67fa07287c7
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/query/plumbing.rs:381
  30:     0x55a98d3354d3 - rustc_query_system::dep_graph::graph::DepGraph<K>::with_task_impl::h11bf9a9e1ca32d34
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/dep_graph/graph.rs:303
  31:     0x55a98d33bc83 - rustc_query_system::dep_graph::graph::DepGraph<K>::with_task::hedbe0ec06ede7893
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/dep_graph/graph.rs:200
  32:     0x55a98d77826b - rustc_query_system::query::plumbing::force_query_with_job::{{closure}}::{{closure}}::h9e28f64858a58b5f
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/plumbing.rs:599
  33:     0x55a98d79b8ed - rustc_middle::ty::query::plumbing::<impl rustc_query_system::query::QueryContext for rustc_middle::ty::context::TyCtxt>::start_query::{{closure}}::{{closure}}::{{closure}}::haa1e8d70b4186d17
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/query/plumbing.rs:72
  34:     0x55a98d7c0225 - stacker::maybe_grow::h7ea9a738343fe635
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.12/src/lib.rs:55
  35:     0x55a98d7c0225 - rustc_data_structures::stack::ensure_sufficient_stack::h095ce5ae6cb51e6e
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_data_structures/src/stack.rs:16
  36:     0x55a98d79b367 - rustc_middle::ty::query::plumbing::<impl rustc_query_system::query::QueryContext for rustc_middle::ty::context::TyCtxt>::start_query::{{closure}}::{{closure}}::hd051512e347679c5
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/query/plumbing.rs:72
  37:     0x55a98d491a76 - rustc_middle::ty::context::tls::enter_context::{{closure}}::h5b7706f6f3c3a087
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1722
  38:     0x55a98d4987c5 - rustc_middle::ty::context::tls::set_tlv::h0e05fd975df1ad1c
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1706
  39:     0x55a98d490a46 - rustc_middle::ty::context::tls::enter_context::h1afbb67ce035486c
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1722
  40:     0x55a98d79a12f - rustc_middle::ty::query::plumbing::<impl rustc_query_system::query::QueryContext for rustc_middle::ty::context::TyCtxt>::start_query::{{closure}}::h0f6c7bd43912d6af
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/query/plumbing.rs:71
  41:     0x55a98d497cdf - rustc_middle::ty::context::tls::with_related_context::{{closure}}::h7b203b14b3ac2593
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1766
  42:     0x55a98d49022f - rustc_middle::ty::context::tls::with_context::{{closure}}::he0421f88231acd0b
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1750
  43:     0x55a98d493afa - rustc_middle::ty::context::tls::with_context_opt::h57a6f3cb3f9bc897
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1739
  44:     0x55a98d48de73 - rustc_middle::ty::context::tls::with_context::hb618c05b5bc30046
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1750
  45:     0x55a98d49682f - rustc_middle::ty::context::tls::with_related_context::h6c0882b171c030b9
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1763
  46:     0x55a98d777991 - rustc_middle::ty::query::plumbing::<impl rustc_query_system::query::QueryContext for rustc_middle::ty::context::TyCtxt>::start_query::h97db9f083543c795
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/query/plumbing.rs:60
  47:     0x55a98d777991 - rustc_query_system::query::plumbing::force_query_with_job::{{closure}}::ha181903f2b291853
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/plumbing.rs:589
  48:     0x55a98d77205a - rustc_query_system::query::plumbing::with_diagnostics::hc8ede9b6162fc5f0
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/plumbing.rs:296
  49:     0x55a98d77205a - rustc_query_system::query::plumbing::force_query_with_job::h72dfe0285a9e118c
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/plumbing.rs:588
  50:     0x55a98d77205a - rustc_query_system::query::plumbing::try_execute_query::ha790bbccaa61b517
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/plumbing.rs:415
  51:     0x55a98d77205a - rustc_query_system::query::plumbing::get_query_impl::{{closure}}::hd6e1cc46a256e5ac
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/plumbing.rs:639
  52:     0x55a98d759d13 - <rustc_query_system::query::caches::DefaultCache<K,V> as rustc_query_system::query::caches::QueryCache>::lookup::h627e3007bc8c764b
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/caches.rs:110
  53:     0x55a98d759d13 - rustc_query_system::query::plumbing::try_get_cached::h1a9c3b5d005cfad9
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/plumbing.rs:369
  54:     0x55a98d759d13 - rustc_query_system::query::plumbing::get_query_impl::h843708b7c17c2068
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/plumbing.rs:631
  55:     0x55a98d4b8ce6 - rustc_query_system::query::plumbing::get_query::hcd51cd0eda2acc9d
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_query_system/src/query/plumbing.rs:731
  56:     0x55a98d4b8ce6 - rustc_middle::ty::query::TyCtxtAt::optimized_mir::h113d5ae55ef8f630
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/query/plumbing.rs:491
  57:     0x55a98d4b8ce6 - rustc_middle::ty::query::<impl rustc_middle::ty::context::TyCtxt>::optimized_mir::hf5329995994d3e38
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/query/plumbing.rs:452
  58:     0x55a98d4b8ce6 - prusti_viper::encoder::encoder::Encoder::encode_stub_pure_function_use::h595f2b3fe73d735c
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/encoder.rs:1691
  59:     0x55a98d6f32ed - <prusti_viper::encoder::pure_function_encoder::PureFunctionBackwardInterpreter as prusti_viper::encoder::mir_interpreter::BackwardMirInterpreter>::apply_terminator::h9f2725f5a281c5af
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/pure_function_encoder.rs:685
  60:     0x55a98d72d4c1 - prusti_viper::encoder::mir_interpreter::run_backward_interpretation::hdb2dd3e79a2bf361
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/mir_interpreter.rs:62
  61:     0x55a98d6ea5d3 - prusti_viper::encoder::pure_function_encoder::PureFunctionEncoder::encode_body::h9058b83a93c49d7e
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/pure_function_encoder.rs:64
  62:     0x55a98d4b52d5 - prusti_viper::encoder::encoder::Encoder::encode_pure_function_body::h392a1334f79aec72
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/encoder.rs:1455
  63:     0x55a98d5dbf93 - prusti_viper::encoder::spec_encoder::SpecEncoder::encode_expression::h74c56ff8766c2a41
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/spec_encoder.rs:351
  64:     0x55a98d5d7bd2 - prusti_viper::encoder::spec_encoder::SpecEncoder::encode_assertion::h6b77266a010aae5b
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/spec_encoder.rs:142
  65:     0x55a98d5d7171 - prusti_viper::encoder::spec_encoder::encode_spec_assertion::hbb21b2c7954b1fcb
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/spec_encoder.rs:66
  66:     0x55a98d4adfa3 - prusti_viper::encoder::encoder::Encoder::encode_assertion::h94a145dbb5d7858a
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/encoder.rs:1104
  67:     0x55a98d6a464c - prusti_viper::encoder::procedure_encoder::ProcedureEncoder::encode_postcondition_expr::hdd5ef6c5f9c0f78a
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/procedure_encoder.rs:3122
  68:     0x55a98d6a8726 - prusti_viper::encoder::procedure_encoder::ProcedureEncoder::encode_postconditions::h3f7f0e4df87cd31e
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/procedure_encoder.rs:3366
  69:     0x55a98d67c58c - prusti_viper::encoder::procedure_encoder::ProcedureEncoder::encode::h6071ba1aa9039aa1
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/procedure_encoder.rs:425
  70:     0x55a98d4ad537 - prusti_viper::encoder::encoder::Encoder::encode_procedure::h68d2ef2b4cadc325
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/encoder.rs:1059
  71:     0x55a98d4b9b2f - prusti_viper::encoder::encoder::Encoder::process_encoding_queue::h844fe4d8147a12c2
                               at /home/aakp/prusti-dev/prusti-viper/src/encoder/encoder.rs:1742
  72:     0x55a98d471f44 - prusti_viper::verifier::Verifier::verify::h20b323fa9e21d483
                               at /home/aakp/prusti-dev/prusti-viper/src/verifier.rs:243
  73:     0x55a98d25c8d2 - prusti_driver::verifier::verify::hd1b65d962c6e5396
                               at /home/aakp/prusti-dev/prusti/src/verifier.rs:48
  74:     0x55a98d25da0f - <prusti_driver::callbacks::PrustiCompilerCalls as rustc_driver::Callbacks>::after_analysis::{{closure}}::h99db1dfeb5e01d14
                               at /home/aakp/prusti-dev/prusti/src/callbacks.rs:82
  75:     0x55a98d273725 - rustc_interface::passes::QueryContext::enter::{{closure}}::hdfe7c6cfbb747326
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_interface/src/passes.rs:722
  76:     0x55a98d275612 - rustc_middle::ty::context::tls::enter_context::{{closure}}::h1081aa1600bf04f4
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1722
  77:     0x55a98d2756a9 - rustc_middle::ty::context::tls::set_tlv::hc1ad8179afee0266
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1706
  78:     0x55a98d2755e6 - rustc_middle::ty::context::tls::enter_context::h6671782c828b5ed8
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_middle/src/ty/context.rs:1722
  79:     0x55a98d2736d8 - rustc_interface::passes::QueryContext::enter::h6a75ff36aff03717
                               at /home/aakp/.rustup/toolchains/nightly-2020-09-30-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/compiler/rustc_interface/src/passes.rs:722
  80:     0x55a98d26f5ac - <prusti_driver::callbacks::PrustiCompilerCalls as rustc_driver::Callbacks>::after_analysis::h835c1b34b92c1817
                               at /home/aakp/prusti-dev/prusti/src/callbacks.rs:48
  81:     0x7f5eabdb8fe6 - rustc_interface::queries::<impl rustc_interface::interface::Compiler>::enter::h18c853e028e6c9f9
  82:     0x7f5eabd4b7c7 - rustc_span::with_source_map::hc6e7fae929044d02
  83:     0x7f5eabdba652 - rustc_interface::interface::create_compiler_and_run::h8ccd8eefc3bc1dbc
  84:     0x7f5eabda049a - scoped_tls::ScopedKey<T>::set::h42e297f49f929a85
  85:     0x7f5eabdbd9b1 - std::sys_common::backtrace::__rust_begin_short_backtrace::he35198b337657261
  86:     0x7f5eabd38b4e - core::ops::function::FnOnce::call_once{{vtable.shim}}::hbc4894a59657eef5
  87:     0x7f5eab4f57ba - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h1080dfe0ef616bdf
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/alloc/src/boxed.rs:1042
  88:     0x7f5eab4f57ba - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hd2747e1f2d5cec32
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/alloc/src/boxed.rs:1042
  89:     0x7f5eab4f57ba - std::sys::unix::thread::Thread::new::thread_start::hd0f336b4ef6808a7
                               at /rustc/381b445ff5751f9f39ec672b623372dff09c276e/library/std/src/sys/unix/thread.rs:87
  90:     0x7f5ea9bd2669 - start_thread
  91:     0x7f5ea9e5d2b3 - __clone
  92:                0x0 - <unknown>

```

Reply:
Is this solved? The meaning of `missing MIR for DefId(2:2102 ~ core[5bc6]::ops::arith::Add::add)` is that Prusti is for some reason trying to access the *implementation* of `core::ops::arith::Add::add`, not a call to it.

Reply:
Ah, good, I just saw the last commit.

Reply:
@fpoli, I'd not say that this issue is completely resolved. I've hit another issue-

```
Verification of 3 items...
error: [Prusti internal error] consistency error: Consistency error: Function equals$__$TY$__m_prusti_contracts$$ghost$$GhostInt$_beg_$_end_$m_prusti_contracts$$ghost$$GhostInt$_beg_$_end_$$bool$ with formal arguments ArraySeq(_left: Ref, _right: Ref) cannot be applied to provided arguments ArraySeq(_0, _1 + _2). (@10.11)

Verification failed
```

@ vakaras suggested that we might need to special-case the `eq` encoding for `GhostInt` similar to `isize`. Currently, I'm looking into this.

Reply:
@cmatheja `GhostInt` is a mathematical integer type that we want to encode in the same way as `isize` just without any bounds. Any suggestions regarding the consistency error?